src/core/stream/period/period_stream.ts

```diff
@@ -191,8 +194,21 @@ export default function PeriodStream({
         }));
 
       return observableConcat<IPeriodStreamEvent>(
-        observableOf(EVENTS.adaptationChange(bufferType, adaptation, period)),
-        newStream$
+        observableOf(EVENTS.adaptationChange(bufferType,
+                                             adaptation,
+                                             period)),
+        newStream$.pipe(mergeMap((evt) => {
+          if (index !== 0 && bufferType === "audio") {
+            isReadyToAdaptAudioSwitchingStrategy = true
+            if (options.audioTrackSwitchingMode === "reload") {
+              return clock$.pipe(mergeMap((tick) => observableOf(EVENTS.needsMediaSourceReload(period, tick))))
+            } else if (evt.type === "added-segment" && isReadyToAdaptAudioSwitchingStrategy) {
+              isReadyToAdaptAudioSwitchingStrategy = false
+              return observableOf(EVENTS.needsSourceBufferFlush(bufferType))
+            }
+          }
+          return observableOf(evt);
+        }))
```

Comment:
Nice!

That's good looking code and I don't see any problem here, but there's two little things that still bother me a little:

  1. Right now, the decisions to reload based on an Adaptation switch is broken up in two places. In `getAdaptationSwitchStrategy` for the video and here for the audio.

      That's a shame because they both do the exact same thing (ask for reload on track switch). A future modification/fix at one place has a chance of not being copied at the other place, which could lead to weird behavior differences. So I think keeping it DRY is better here. 

  2. I like the spirit of the flush logic here but I feel it could be done better (with another Observable and playing with operators?) without the need for the `isReadyToAdaptAudioSwitchingStrategy` boolean (whose name may indicate that we are doing an ugly thing - as in difficult to properly explain).

      I may be wrong here, so do not hesitate to tell me if no alternative seems better.

Reply:
1. Ok, I see, It should be done in the last commit pushed.

2. Yes, I was wondering if observable could be used to do the trick here. I will check if we can do it with observable.

Reply:
2. Ok! I got a working version with an Observable logic, after battling, I find the new observable approach cleaner as you mentioned it.

What do you think?